srcjs/output_binding_html.js

```diff
@@ -107,23 +107,68 @@ function renderDependency(dep) {
   }
 
   if (dep.stylesheet) {
-    var stylesheets = $.map(asArray(dep.stylesheet), function(stylesheet) {
-      var this_href = href + "/" + encodeURI(stylesheet);
-      // If a styleSheet with this href is already active, then disable it
-      var re = new RegExp(this_href + "$");
-      for (var i = 0; i < document.styleSheets.length; i++) {
-        var h = document.styleSheets[i].href;
-        if (!h) continue;
-        if (restyle && h.match(re)) document.styleSheets[i].disabled = true;
+    var links = $.map(asArray(dep.stylesheet), function(stylesheet) {
+      return $("<link rel='stylesheet' type='text/css'>").attr("href", href + "/" + encodeURI(stylesheet));
+    });
+
+    if (!restyle) {
+      $head.append(links);
+    } else {
+      $.map(links, function(link) {
+        // Find any document.styleSheets that match this link's href
+        // so we can remove it after bringing in the new stylesheet
+        var oldSheet = findSheet(link.attr("href"));
+        // Add a timestamp to the href to prevent caching
+        var href = link.attr("href") + "?restyle=" + new Date().getTime();
+        // Use inline <style> approach for IE, otherwise use the more elegant
+        // <link> -based approach
+        if (isIE()) {
+          refreshStyle(href, oldSheet);
+        } else {
+          link.attr("href", href);
+          link.attr("onload", () => {
+            setTimeout(() => removeSheet(oldSheet), 10);
+          });
+          $head.append(link);
+        }
+      });
+
+      // This inline <style> based approach works for IE11
+      function refreshStyle(href, oldSheet) {
+        var xhr = new XMLHttpRequest();
+        xhr.open('GET', href);
+        xhr.onload = function() {
+          var id = "shiny_restyle_" + href.split("?restyle")[0].replace(/\W/g, '_');
+          var oldStyle = $head.find("style#" + id);
+          var newStyle = $("<style>").attr("id", id).html(xhr.responseText);
+          $head.append(newStyle);
+          if (oldStyle) setTimeout(() => oldStyle.remove(), 10);
```

Comment:
What if `oldStyle` doesn't exist?

Reply:
Even if the `.find()` comes up empty, `oldStyle.remove()` won't error out since it's a jQuery selection, so I suppose we could either remove the `if` or change it to `if (oldStyle.length > 0) setTimeout(() => oldStyle.remove(), 10);`?